### PR TITLE
Detect Windows-1252 encoding

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: sonar-scanner-engine/src/test/java/org/sonar/scanner/scan/filesystem/ByteCharsetDetectorTest.java [hunk 1]
```diff
@@ -132,4 +133,10 @@ public void invalidBOM() {
     byte[] b3 = new byte[0];
     assertThat(charsets.detectBOM(b3)).isNull();
   }
+
+  @Test
+  public void windows1252() throws IOException, URISyntaxException {
+    ByteCharsetDetector detector = new ByteCharsetDetector(new CharsetValidation(), StandardCharsets.UTF_8);
+    assertThat(detector.detect(readFile("windows-1252"))).isEqualTo(Charset.forName("Windows-1252"));
```
Comment (henryju) [inline sonar-scanner-engine/src/test/java/org/sonar/scanner/scan/filesystem/ByteCharsetDetectorTest.java]: Does Charset.forName("Windows-1252") always returns something, even on Linux?

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/CharsetValidation.java [hunk 2]
```diff
@@ -254,6 +254,31 @@ private static int read16bit(byte[] buffer, int i, boolean le) {
...
+  /**
+   * Verify that the buffer doesn't contain bytes that are not supposed to be used by Windows-1252.
+   *
+   * @return Result object with Validation.MAYBE and Windows-1252 if no unknown characters are used,
+   * otherwise Result.INVALID
+   * @param buf byte buffer to validate
+   */
+  public Result isValidWindows1252(byte[] buf) {
+    for (byte b : buf) {
+      switch (b) {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/CharsetValidation.java]: CRITICAL Add a default case to this switch. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/CharsetValidation.java [hunk 3]
```diff
@@ -254,6 +266,27 @@ private static int read16bit(byte[] buffer, int i, boolean le) {
...
+   * Verify that the buffer doesn't contain bytes that are not supposed to be used by Windows-1252.
+   *
+   * @return Result object with Validation.MAYBE and Windows-1252 if no unknown characters are used,
+   * otherwise Result.INVALID
+   * @param buf byte buffer to validate
+   */
+  public Result isValidWindows1252(byte[] buf) {
+    for (byte b : buf) {
+      if (!VALID_WINDOWS_1252[b + 128]) {
+          return Result.INVALID;
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/CharsetValidation.java]: MAJOR Make this line start at column 9. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request closed without merging**